src/main/kotlin/brs/db/sql/SqlTransactionDb.kt

```diff
@@ -140,37 +140,63 @@ internal class SqlTransactionDb(private val dp: DependencyProvider) : Transactio
-                        if (transaction.referencedTransactionFullHash != null) setReferencedTransactionFullhash(*transaction.referencedTransactionFullHash) // TODO better way of setting
-                        height = transaction.height
-                        blockId = transaction.blockId
-                        if (transaction.signature != null) setSignature(*transaction.signature!!) // TODO better way of setting
-                        timestamp = transaction.timestamp
-                        type = transaction.type.type
-                        subtype = transaction.type.subtype
-                        senderId = transaction.senderId
-                        val attachment = getAttachmentBytes(transaction)
-                        if (attachment != null) setAttachmentBytes(*attachment) // TODO better way of setting
-                        blockTimestamp = transaction.blockTimestamp
-                        setFullHash(*transaction.fullHash) // TODO better way of setting
-                        version = transaction.version
-                        hasMessage = transaction.message != null
-                        hasEncryptedMessage = transaction.encryptedMessage != null
-                        hasPublicKeyAnnouncement = transaction.publicKeyAnnouncement != null
-                        hasEncrypttoselfMessage = transaction.encryptToSelfMessage != null
-                        ecBlockHeight = transaction.ecBlockHeight
-                        if (transaction.ecBlockId != 0L) ecBlockId = transaction.ecBlockId
-                        changed(true)
-                    }
-                }).execute()
+            dp.db.useDslContext { ctx ->
+                var insertQuery = ctx.insertInto(
+                    TRANSACTION,
+                    TRANSACTION.ID,
+                    TRANSACTION.DEADLINE,
+                    TRANSACTION.SENDER_PUBLIC_KEY,
+                    TRANSACTION.RECIPIENT_ID,
+                    TRANSACTION.AMOUNT,
+                    TRANSACTION.FEE,
+                    TRANSACTION.REFERENCED_TRANSACTION_FULLHASH,
+                    TRANSACTION.HEIGHT,
+                    TRANSACTION.BLOCK_ID,
+                    TRANSACTION.SIGNATURE,
+                    TRANSACTION.TIMESTAMP,
+                    TRANSACTION.TYPE,
+                    TRANSACTION.SUBTYPE,
+                    TRANSACTION.SENDER_ID,
+                    TRANSACTION.ATTACHMENT_BYTES,
+                    TRANSACTION.BLOCK_TIMESTAMP,
+                    TRANSACTION.FULL_HASH,
+                    TRANSACTION.VERSION,
+                    TRANSACTION.HAS_MESSAGE,
+                    TRANSACTION.HAS_ENCRYPTED_MESSAGE,
+                    TRANSACTION.HAS_PUBLIC_KEY_ANNOUNCEMENT,
+                    TRANSACTION.HAS_ENCRYPTTOSELF_MESSAGE,
+                    TRANSACTION.EC_BLOCK_HEIGHT,
+                    TRANSACTION.EC_BLOCK_ID
+                )
+                transactions.map { transaction ->
```

Comment:
`forEach` would be better as it doesn't create a new collection with the map results